Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -141,6 +141,12 @@ public class DocumentIterator: IteratorProtocol {
         return nil
     }
 
+    internal func overwriteCurrentValue(with newValue: Overwritable) throws {
+        precondition(newValue.bsonType == self.currentType,
+                     "Expected \(newValue) to have BSON type \(self.currentType), but has type \(newValue.bsonType)")
```

Comment:
Correct me if I'm wrong, but this is just a sanity check to ensure one Overwritable (calling this method) isn't trying to overwrite itself with a different Overwritable type. Is that right?

And this ties in with the logic before this method is called in `setValue()`:

```
if let ov = newValue as? Overwritable, ov.bsonType == existingType
```

Reply:
yes, exactly, just a sanity check. I don't expect we will ever call this method from anywhere besides the current call site.